cosign/verifier.go

```diff
@@ -0,0 +1,549 @@
+
+import (
+	"context"
+	"crypto"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/notaryproject/ratify-go"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
+	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
+	protorekor "github.com/sigstore/protobuf-specs/gen/pb-go/rekor/v1"
+	"github.com/sigstore/sigstore-go/pkg/bundle"
+	"github.com/sigstore/sigstore-go/pkg/root"
+	"github.com/sigstore/sigstore-go/pkg/tuf"
+	"github.com/sigstore/sigstore-go/pkg/verify"
+	"github.com/sigstore/sigstore/pkg/signature"
+)
+
+const (
+	verifierTypeCosign        = "cosign"
+	signatureArtifactCosign   = "application/vnd.dev.cosign.artifact.sig.v1+json"
+	sigstoreBundleMediaType01 = "application/vnd.dev.sigstore.bundle+json;version=0.1"
+	annotationKeyBundle       = "dev.sigstore.cosign/bundle"
+	AnnotationKeyCert         = "dev.sigstore.cosign/certificate"
+	annotationKeySignature    = "dev.cosignproject.cosign/signature"
+	mediaTypeSimpleSigning    = "application/vnd.dev.cosign.simplesigning.v1+json"
+)
+
+// PublicKeyConfig contains the configuration for key-based verification.
+type PublicKeyConfig struct {
+	// PublicKey for key-based verification.
+	PublicKey crypto.PublicKey
+
+	// SignatureAlgorithm defines the algorithm used for signature verification.
+	// Optional. If not provided, defaults to SHA256.
+	SignatureAlgorithm crypto.Hash
+
+	// ValidityPeriodStart defines the start time for the public key validity
+	// period. Optional. If not provided, the key is considered valid from the
+	// beginning of time.
+	ValidityPeriodStart time.Time
+
+	// ValidityPeriodEnd defines the end time for the public key validity period.
+	// Optional. If not provided, the key is considered valid until the end of
+	// time.
+	ValidityPeriodEnd time.Time
+}
+
+// VerifierOptions contains the options for creating a new Cosign verifier.
+type VerifierOptions struct {
+	// Name is the instance name of the verifier to be created. Required.
+	Name string
+
+	// TrustedRoot defines the trusted root for verification. For keyless verification,
+	// this will contain Fulcio certificates and Rekor public keys. Optional.
+	// If not provided, the default Sigstore trusted root will be used.
+	TrustedRoot *root.TrustedRoot
+
+	// PublicKeyConfig contains the public key and its validity period for key-based verification.
+	// If provided, key-based verification will be performed. Optional.
+	PublicKeyConfigs []*PublicKeyConfig
+
+	// IdentityPolicies contains policies for keyless verification.
+	// These policies specify which OIDC identities are trusted. Optional.
+	IdentityPolicies []verify.PolicyOption
+
+	// IgnoreTlog when set to true, skips Artifact transparency log verification.
+	// Only applies to keyless verification. Optional, defaults to false.
+	IgnoreTlog bool
```

Comment:
It's `Tlog` or `TLog`?

Reply:
I guess it is `TLog`.

Reply:
example in sigstore-go uses Tlog

Reply:
to be consistent in the implementation, I'll change to TLog